src/serde_util.rs

```diff
@@ -67,3 +67,62 @@ where
         (self.parse_fn)(v).map_err(E::custom)
     }
 }
+
+macro_rules! impl_tuple_visitor {
+    ($thing:ident, $len:expr) => {
+        /// Visitor type for de/ser a 32 byte sequence.
+        #[cfg(any(feature = "std", feature = "alloc"))]
+        pub struct $thing<F> {
```

Comment:
I suggest `pub(crate)`, to not leak this by accident.

Reply:
Good idea.